File: backend/src/main/java/io/github/artemy/osipov/shop/service/candy/CandyCommandHandler.java
package io.github.artemy.osipov.shop.service.candy;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import io.github.artemy.osipov.shop.exception.EntityNotFoundException;

@Service
@RequiredArgsConstructor
public class CandyCommandHandler {

    private final CandyRepository repository;

    public void on(CreateCandyCommand event) {
        Candy candy = new Candy();
        candy.setId(event.getId());
        candy.setName(event.getName());
        candy.setFirm(event.getFirm());
        candy.setPrice(event.getPrice());
        candy.setOrder(event.getOrder());

        repository.save(candy);
    }

    public void on(UpdateCandyCommand event) {
        Candy candy = repository.findById(event.getId())
                .orElseThrow(() -> new EntityNotFoundException(Candy.class, event.getId()));

        candy.setName(event.getName());
        candy.setFirm(event.getFirm());
        candy.setPrice(event.getPrice());
        candy.setOrder(event.getOrder());

        repository.save((candy));
    }

    public void on(HideCandyCommand event) {
        Candy candy = repository.findById(event.getId())
                .orElseThrow(() -> new EntityNotFoundException(Candy.class, event.getId()));

        candy.hide();

        repository.save((candy));
    }
}

File: backend/src/main/kotlin/io/github/artemy/osipov/shop/service/candy/CandyCommandHandler.kt
package io.github.artemy.osipov.shop.service.candy

import io.github.artemy.osipov.shop.service.candy.CandyRepository.Companion.getById
import kotlinx.coroutines.reactor.awaitSingle
import org.springframework.stereotype.Service

@Service
class CandyCommandHandler(
    private val repository: CandyRepository
) {
    suspend fun on(event: CreateCandyCommand) {
        val candy = Candy(
            id = event.id,
            name = event.name,
            firm = event.firm,
            price = event.price,
            order = event.order
        )
        repository.add(candy)
    }

    suspend fun on(event: UpdateCandyCommand) {
        val candy = repository.getById(event.id)
        candy.apply {
            name = event.name
            firm = event.firm
            price = event.price
            order = event.order
        }
        repository.save(candy)
    }

    suspend fun on(event: HideCandyCommand) {
        val candy = repository.getById(event.id)
        candy.hide()
        repository.save(candy)
    }
}